samples/wifi/scan/src/main.c

```diff
@@ -195,6 +195,19 @@ static int wifi_scan(void)
 	k_sem_take(&scan_sem, K_MSEC(SCAN_TIMEOUT_MS));
 #endif
 
+#if defined(CONFIG_WIFI_MGMT_FORCED_PASSIVE_SCAN) || defined(CONFIG_WIFI_SCAN_TYPE_PASSIVE)
+	params.scan_type = WIFI_SCAN_TYPE_PASSIVE;
+	if (net_mgmt(NET_REQUEST_WIFI_SCAN, iface, &params,
+				sizeof(struct wifi_scan_params))) {
+		LOG_ERR("Scan request failed");
+
+		return -ENOEXEC;
+	}
+	printk("Passive scan requested\n");
+
+	k_sem_take(&scan_sem, K_MSEC(SCAN_TIMEOUT_MS));
+#endif
```

Comment:
the conditions are only for assigning `params.scan_type` sending `net_mgmt(SCAN)` is common to all.

Reply:
@krish2718 Plan is to do mutiple scans. If we do as you said, we have to call wifi_scan() ,multiple times from main() with specific parameter?

Can I continue like this ? or Should I change to that way(calling mutlitple time from main())?

Reply:
yes, the comment is about avoiding code duplication, it can be implemented in many ways, I would have used a data driven style that defines all combinations and then keep `wifi_scan` generic looping through all combinations.